beta_rec/models/ngcf.py

```diff
@@ -4,6 +4,8 @@
 import torch.nn.functional as F
 from beta_rec.models.torch_engine import Engine
 
+DEVICE = torch.device("cuda" if torch.cuda.is_available() else "cpu")
```

Comment:
It is better to use the self.config["device_str"] to pass the device information across all the program.  The config["device_str"] is initially obtained from the train_engine.py, and then pass it to each XXModelEngine. So you can directly use self.config["device_str"] to get the device string (e.g. "cpu" or "cuda:0").